test/loadtest.js

```diff
@@ -0,0 +1,92 @@
+'use strict'
+
+const http = require('http')
+const {test} = require('tap')
+const {createLogger} = require('../index.js')
+
+const TOTAL_LINES = 100000
+let httpCalls = 0
+let linesReceived = 0
+let sendEvents = 0
+let errorEvents = 0
+
+const server = http.createServer((req, res) => {
+  let buffer = ''
+  req.on('data', (data) => {
+    buffer += data
+  })
+  req.on('end', () => {
+    const body = JSON.parse(buffer)
+    if (++httpCalls % 2 === 0) {
+      res.statusCode = 500
+      res.end('POOF - The request should be retried')
+      return
+    }
+    linesReceived += body.ls.length
+    res.statusCode = 200
+    res.end('Ingested')
+  })
+})
+
+test('Load test to ensure no data loss and expected payloads', async (t) => {
+  const line = 'x'.repeat(500)
+
+  t.test('Start a local HTTP server', (tt) => {
+    server.listen(0)
+    server.on('listening', tt.end)
+  })
+
+  t.test(`Test ${TOTAL_LINES} lines`, (tt) => {
+    const logger = createLogger('<YOUR KEY HERE>', {
+      retryTimes: 100
+    , baseBackoffMs: 100
+    , maxBackoffMs: 500
+    , url: `http://localhost:${server.address().port}`
+    })
+
+    logger.on('send', (evt) => {
+      sendEvents++
+      tt.match(evt, {
+        httpStatus: 200
+      , firstLine: line
+      , lastLine: line
+      , totalLinesSent: Number
+      , totalLinesReady: Number
+      , bufferCount: Number
+      }, 'send event payload is correct')
+    })
+
+    logger.on('error', (evt) => {
+      errorEvents++
+      tt.match(evt, {
```

Comment:
deepEqual

Reply:
eh, same answer.  It fails because I don't have `message` or `name`.  Here, I'm not looking to test those things, so using deep comparison isn't necessary. I can add them though.

In general, each test should test a different thing.  So, if I change that to a `TypeError`, it should fail 1 test for sure, but failing the `loadtest` wouldn't have anything to do with the load test, so it could be left out of that check.

Reply:
Added